astropy/utils/iers/iers.py

```diff
@@ -442,6 +442,36 @@ class because it has different requirements.
                 warn(msg, IERSDegradedAccuracyWarning)
             # No IERS data covering the time(s) and user is OK with no warning.
 
+    def _lagrange_interp(self, x_vals, y_vals, xint):
+        val = np.zeros((y_vals.shape[1:]), dtype=y_vals.dtype)
+
+        for i in range(len(x_vals)):
+            term = y_vals[i]
+            for j in range(len(x_vals)):
+                if i != j:
+                    diff = x_vals[i] - x_vals[j]
+                    diff = np.where(diff == 0, 1e-10, diff)
+                    term *= (xint - x_vals[j]) / diff
+            val += term
+
+        return val
+
+    def _ut1_utc_interp(self, i0, i1, mjd):
+        indices = np.array([i0 - 1, i0, i1, i1 + 1])
+        indices = np.clip(indices, 0, len(self) - 1)
+
+        mjds = self["MJD"][indices].value
+        vals = self["UT1_UTC"][indices].value
+
+        d_vals = np.diff(vals, axis=0)
+        leap_diff = np.cumsum(np.round(d_vals), axis=0)
+        vals[1:] += leap_diff
+
+        val = self._lagrange_interp(mjds, vals, mjd)
+        val -= leap_diff[0]
```

Comment:
I think in the general case you cannot be sure index 0 is the right one to subtract. Which one it is depends on how the 3 surrounding points are selected, so this likely is an array.